joomla/joomla-cms
# New DateTime picker (replaces calendar)

#### UX UI improvements
#### Summary of Changes
- Bootstrapified
- Remove old code for IE<8
- New simplified popup code
- The internals of the date calculations are unchanged
- Base on #707 improved support for non Gregorian calendars
- New time picker code based on select elements (24 or 12h based)
- options to display or not: week numbers, today button, year month in one row, days from previous next month (as disabled)
- Everything is controllable through the calendar field definition (documentation will be done later on)
#### Preview

![screen shot 2016-07-15 at 01 54 13](https://cloud.githubusercontent.com/assets/3889375/16858420/578c67ee-4a2f-11e6-9224-2b0db517294a.png)

With time picker
![screen shot 2016-07-17 at 00 43 45](https://cloud.githubusercontent.com/assets/3889375/16897377/dc174284-4bb7-11e6-970e-1631fdaaa160.png)

Basic view
![screen shot 2016-07-15 at 02 11 02](https://cloud.githubusercontent.com/assets/3889375/16858799/41cc42b4-4a32-11e6-9fb9-8b29569dba47.png)
#### Testing Instructions

You need a 3.7 installation
apply this patch
try various forms (contact, banner, article) and check if the calendar works ok.
edit `/Users/dimitris/Documents/github_projects/joomla1/administrator/components/com_content/models/forms/article.xml` 

``` xml
        <field name="publish_up"
            type="calendar"
            label="COM_CONTENT_FIELD_PUBLISH_UP_LABEL"
            description="COM_CONTENT_FIELD_PUBLISH_UP_DESC"
            format="%Y-%m-%d %H:%M:%S" size="22"
            filter="user_utc"
            todaybutton="true"
            weeknumbers="false"
            timeformat="24"
            showtime="true"
            filltable="true"
            singleheader="false"
            minyear="-100"
            maxyear="+50" />
```

and alter different options to check if everything still works ok

Install Persian language and test if that calendar works ok!

Try some 3rd PD components and check if they still functioning ok.

Report any problems
